### ISSUE TITLE
filter 'delete_ind is null' needs to be added to these views in PG to exclude the committees that have been deleted

### ISSUE THREAD

**Issue by hcaofec** [18F/openFEC#2627]
public.fec_vsum_f1_vw
public.fec_vsum_f2_vw
public.fec_vsum_f3_vw
public.fec_vsum_f3p_vw
public.fec_vsum_f3ps_vw
public.fec_vsum_f3s_vw
public.fec_vsum_f3x_vw
public.fec_vsum_f3z_vw
public.fec_vsum_f7_vw
public.fec_vsum_f9_vw

- [ ] change SQL
- [ ] determine if views have dependencies

**Comment by LindsayYoung** [comment]
Thanks @hcaofec! 

This ticket brings to mind that we moved off of the vsum tables for totals. I am wondering if we should not be using them to build our reports views too. 

@fecjjeng Am I remembering correctly that vsum don't handle amendments well? Was that just for totals or is there a case where we should be using vsum? I would like to be consistent about our base tables for reports and totals if possible. Thanks!

**Comment by fecjjeng** [comment]
As far as I can remember, the vsum views did NOT handle amendment for schedules well.  It is correct for forms.  I am at the airport right now and has no access to FEC database, so please double check with @paul or Jerry to make sure.

Having said that, it would be a good idea to use the base tables for the reports.  I remember we had this discussion before with you and Carlo to remove the vsum views and keep the view definition (with the delete_ind is null filter added, of course) in a reference table to be used as a template to lower the interdependencies but keep the consistency.

Jean

On Aug 29, 2017, at 10:26 AM, Lindsay Young <notifications@github.com<mailto:notifications@github.com>> wrote:


Thanks @hcaofec<https://github.com/hcaofec>!

This ticket brings to mind that we moved off of the vsum tables for totals. I am wondering if we should not be using them to build our reports views too.

@fecjjeng<https://github.com/fecjjeng> Am I remembering correctly that vsum don't handle amendments well? Was that just for totals or is there a case where we should be using vsum? I would like to be consistent about our base tables for reports and totals if possible. Thanks!

**Comment by jwchumley** [comment]
This is currently a production issue causing incorrect rows to appear on fec.gov. How do we move forward and make the changes so the deleted rows don't stay on the website? 

As for the issues with the v_sum views if I recall that was because the views were using the v_sum table to limit to the correct rows but displaying data columns from the form views. I thought we changed them to use the v_sum columns which do handle amendments correctly. 

Was there something else?

@LindsayYoung @fecjjeng

**Comment by fecjjeng** [comment]
For the schedules views, we already switch to use the fItem instead of vsum, since vsum does not handle the schedule data related to the amendment.
For the form data, vsum views are fine with the amendment data.  We display all historical form data, the rows existing in vsum table will get a "label" of most_recent_filing_flsg='Y'.  The problem now is that in several earlier vsum views that were created when NML form tables were transferred by Java programs, we did not include the "delete_ind is null" filter since the logic had already been handled so those rows would not exist in the nml tables.  However, after switching to use the goldengate to transfer the NML tables (which include ALL data, even the rows with delete_ind set), we did not go back to those earlier views to add the delete_ind is null filter.  The vsum views created later do not have these problem.

Updating the vsum form views itself is straightforward.  The problem is that some of the vsum views with this problem have materialized views built on top of them, to change the view definition will require dropping and recreating those dependent materialized views.  That's why Helen was asking the question in Slack how should we proceed with the change without impact the production site.

Jean

On Sep 5, 2017, at 11:41 AM, Jeff   <notifications@github.com<mailto:notifications@github.com>> wrote:


This is currently a production issue causing incorrect rows to appear on fec.gov<http://fec.gov>. How do we move forward and make the changes so the deleted rows don't stay on the website?

As for the issues with the v_sum views if I recall that was because the views were using the v_sum table to limit to the correct rows but displaying data columns from the form views. I thought we changed them to use the v_sum columns which do handle amendments correctly.

Was there something else?

@LindsayYoung<https://github.com/lindsayyoung> @fecjjeng<https://github.com/fecjjeng>

**Comment by fecjjeng** [comment]
@ccostino @LindsayYoung For these 10 fec views that need to be updated, 4 of them has ofec materialized views dependencies from database point of view.  My understanding is that users/API only access the ofec materialized views, not the fec views, is it correct?  
If that is the case, the 6 fec views that are not have ofec materialized views dependencies can be simply recreated with the new definition.  For those 4 fec views with ofec materialized views dependencies, my previous experience was that the depending ofec materialized views need to be recreated.  However, I just realized that since this time we only update the query definition, no output columns changes, we can just update the fec views definition as well without impacting the ofec materialized views depending on them.  If I took this approach, will any any procedures/programs be impacted ?  Any files in GitHub need to be updated?

**Comment by LindsayYoung** [comment]
I think we can recreate the views on our end without too much trouble. We can verify by doing this in dev. I think we want to coordinate when we do this on production so that we can recreate our views as soon as you recreate your view.

**Comment by fecjjeng** [comment]
@LindsayYoung @lindsay @ccostino Just want to make sure we are on the same page completely.  Please let me know if I miss something.
1. There is NO direct access to the base fec_%vw from users/APIs.
2. Since there is no ofec materialized views definition changed in this case, there is NO need to update any files in GitHub.
3. ofec materialized views DATA refresh are needed AFTER base view definition updated.   Update_schema is the script that needed to be run to refresh all the ofec materialized views.
4. We will test this out by:
      Step 1: I will update the fec views definition in DEV environment and let you know.  
      Step 2: You will run the update_schema in DEV (could you let me know when you do so?  I would like to see how you call this the update_schema if that is OK with you)

**Comment by fecjjeng** [comment]
@LindsayYoung @lindsay @ccostino I had updated the definition of these 10 fec views in DEV environment.  As expected in this case, I did not encountered errors that prevent me from making the updates due to the depending ofec materialized views since no output column changes in the base views.  The depending ofec materialized views will not change, the data in depending ofec materialized views  will be refreshed AFTER update_schema run (either you run it manually or by the nightly process you have in place).  We can verify it tomorrow.
